src/lib.rs

```diff
@@ -33,6 +33,25 @@ pub fn start_logger() {
     env_logger::init_from_env(env);
 }
 
+pub async fn run_mock_server(
+    http_bind: String,
+    admin_http_bind: String,
+    state: Arc<RwLock<State>>,
+    close_web_channel: tokio::sync::oneshot::Receiver<()>,
+    close_admin_channel: tokio::sync::oneshot::Receiver<()>,
+) -> Result<(), Error> {
+    let server = run_web_server(http_bind, state.clone(), close_web_channel);
+    let admin_server = run_admin_server(admin_http_bind, state.clone(), close_admin_channel);
+
+    let (web, admin) =
+        tokio::try_join!(tokio::task::spawn(server), tokio::task::spawn(admin_server))?;
+    match (web, admin) {
+        (Err(e), _) => Err(anyhow::anyhow!(e)),
```

Comment:
Maybe more use a `context` on error than creqteq  new error from the original one